### PR TITLE
Fix unit test failure for FIPS 140-2 + WOLFSSL_ARMASM

### PR DESCRIPTION
**Pull request by lealem47**
# Description

Fix unit test failure for FIPS 140-2 + WOLFSSL_ARMASM by comparing return val to expected error code

Fixes zd#17472

# Testing

Tested with `./configure --enable-fips=v2 --enable-armasm` on the wolfssl-5.6.6-commercial-fips-ARMv8-G-v2 bundle. Make check will pass with this commit.

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: tests/api.c [hunk 1]
```diff
@@ -18498,7 +18498,11 @@ static int test_wc_AesGcmEncryptDecrypt(void)
         BAD_FUNC_ARG);
     ExpectIntEQ(wc_AesGcmDecrypt(&aes, dec, enc, sizeof(enc)/sizeof(byte), iv,
         sizeof(iv)/sizeof(byte), resultT, sizeof(resultT) + 1, a, sizeof(a)),
+    #if (defined(HAVE_FIPS) && FIPS_VERSION_LE(2,0) && defined(WOLFSSL_ARMASM))
+        AES_GCM_AUTH_E);
```
Comment (dgarske) [inline tests/api.c:18502]: Prefer the trailing `);` is outside the `#if` block.

Comment (dgarske) [reply tests/api.c:18502]: I might also be required you do this check outside `ExpectIntEQ` since that is also a macro.

**Review by dgarske — CHANGES_REQUESTED** [review]

File: tests/api.c (hunk 1 above)
Comment (lealem47) [reply tests/api.c:18502]: Done

**Review by dgarske — APPROVED** [review]

**Comment by dgarske** [comment]
Retest this please

**Pull request merged**